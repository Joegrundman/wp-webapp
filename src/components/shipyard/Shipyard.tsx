import ArrowButton from 'Atoms/arrowbutton/ArrowButton';
import * as React from 'react'
import { getGame } from 'Wp/Game'
import locals from './Shipyard.css'

interface IShipyardState {
  currentShipyard: number
}

const shipyards = [
  'Atlantic',
  'Glasgow',
  'Pacific',
  'Tokyo',
  'Kiel',
  'Genoa',
  'Trieste',
  'Lorient',
  'Marseilles',
  'Leningrad',
  'Sevastopol',
  'Canada',
  'Australia'
]

class Shipyard extends React.Component<{}, IShipyardState> {
  public syBgContext: CanvasRenderingContext2D | null
  public syContext: CanvasRenderingContext2D | null

  constructor(props: any){

    super(props)
    this.state = {
      currentShipyard: 0
    }

    this.syBgContext = null
    this.syContext = null
  }

  public componentDidMount (): void {
    const syBgCanvas: HTMLCanvasElement = document.getElementById('shipyardCanvasBackground') as HTMLCanvasElement
    this.syBgContext = syBgCanvas.getContext('2d') as CanvasRenderingContext2D
    const syCanvas: HTMLCanvasElement = document.getElementById('shipyardCanvas') as HTMLCanvasElement
    this.syContext = syCanvas.getContext('2d') as CanvasRenderingContext2D
    this.drawShipyardBackground();
    getGame().setSelectedShipyard(shipyards[this.state.currentShipyard], this.syContext)
  }

  public drawShipyardBackground (): void {
    if(!this.syBgContext) {
      return
    }
    this.syBgContext.strokeStyle= '#cdcdcd'
    this.syBgContext.lineWidth = 1

    for(let i: number = 28.5; i < 330; i += 68 ) {
      this.syBgContext.beginPath()
      this.syBgContext.moveTo(i, 0)
      this.syBgContext.lineTo(i, 368)
      this.syBgContext.closePath()
      this.syBgContext.stroke()
    }

    const seasons: string[] = ['  spring', 'summer', '    fall', '  winter']
    this.syBgContext.font = "15px sans-serif"
    this.syBgContext.fillStyle = '#cdcdcd'

    seasons.forEach((season: string, i: number): void => {
      if(!this.syBgContext) { return }
      this.syBgContext.fillText(season, 36 + (68 * i), 18)
    })

    for (let i: number = 28.5; i < 410; i += 68) {
      this.syBgContext.beginPath()
      this.syBgContext.moveTo(0, i)
      this.syBgContext.lineTo(300, i)
      this.syBgContext.closePath()
      this.syBgContext.stroke()
    }

    const levels: string[] = ['5', '4', '3', '2', '1']
    this.syBgContext.font = "18px sans-serif"
    levels.forEach((level: string, i: number): void => {
      if(!this.syBgContext) { return }
      this.syBgContext.fillText(level, 8, 68 + (68 * i))
    })

    this.syBgContext.font = "15px sans-serif"
    this.syBgContext.fillText('Waiting for repair', 100, 400)
  }

  public decrementCurrentShipyard = () => {
    const currentShipyard = this.state.currentShipyard === 0 ?
      shipyards.length -1 :
      this.state.currentShipyard - 1
    
    this.setState({
      currentShipyard
    },
    () => getGame().setSelectedShipyard(shipyards[this.state.currentShipyard], this.syContext as CanvasRenderingContext2D ))
  }

  public incrementCurrentShipyard = () => {
    const currentShipyard = this.state.currentShipyard < shipyards.length - 1 ?
      this.state.currentShipyard + 1 :
      0

    this.setState({
      currentShipyard
    },
    () => getGame().setSelectedShipyard(shipyards[this.state.currentShipyard], this.syContext as CanvasRenderingContext2D))
  }

  public render (): JSX.Element {

    const shipyardCanvasSize = {
      height: 470,
      width: 300
    }
    return (
      <div>
        <h1 className={locals.header}>Shipyards</h1>
        <div className={locals.subheaderWithArrows}>
          <ArrowButton direction="left" action={this.decrementCurrentShipyard}/>
          <h2 className={locals.subheaderCenterText}>{shipyards[this.state.currentShipyard]}</h2>
          <ArrowButton direction="right" action={this.incrementCurrentShipyard}/>
        </div>
        <div className={locals.canvasContainer} style={{width: shipyardCanvasSize.width}}>
          <canvas
            className={locals.shipyardCanvas}
            id="shipyardCanvasBackground"
            height={shipyardCanvasSize.height}
            width={shipyardCanvasSize.width}/>
          <canvas
            className={locals.shipyardCanvas}
            id="shipyardCanvas"
            height={shipyardCanvasSize.height}
            width={shipyardCanvasSize.width}/>
        </div>

      </div>
    )
  }
}

export default Shipyard
