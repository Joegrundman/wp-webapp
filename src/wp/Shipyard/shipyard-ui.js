// WP.Shipyard.UI = {
// 	addYardToDropdown: function (yrd) {
// 		if (!(yrd.name == "Captured Shipyard")){$('#syYard').append($("<option />").attr("value", yrd.id).text(yrd.name)); }
// 	},

// 	fillShipyardList: function () {
// 		$('#syYard').empty();
// 		var shipyards = game.shipyards;
// 		for (var i = 0; i < shipyards.length; i++) {
// 			WP.Shipyard.UI.addYardToDropdown(shipyards[i]);
// 		}
// 		$("#syYard").prop("selectedIndex", 11);
// 		shipyard.handleShipyardSelected($('#syYard').val());
// 	},

// 	handleMenuButton: function () {

// 		$("#shipyard").dialog('open');
// 		$("#shipyard").dialog("option", "width", 367);
// 		$("#shipyard").dialog("option", "resizable", false);
// 		WP.Shipyard.UI.fillShipyardList();
// 	}
// }

// WP.Shipyard.Mouse = {}

// // WP.Shipyard.prototype.draw = function () {
// // 	var syd = game.getShipyards($('#syYard').val());
// // 	var cty = game.getCountryFromName(syd.owner);
// // 	var units = new Array();

// // 	var holder = WP.UnitHolder.unitHolderBuilder(shipyardCtx, $("#syDetails"));
// // 	// var holder = WP.UnitHolder.Util.unitHolderBuilder(shipyardCtx, $("#syDetails"));

// // 	for (var i = 0; i < syd.shipyardUnits.length; i++) {
// // 		var syUnit = syd.shipyardUnits[i];
// // 		var unit = game.getUnitForShipyard(syUnit.id, syUnit.x, syUnit.y);
// // 		units[units.length] = unit;
// // 	}

// // 	holder.units = units;
// // 	holder.drawShipyard();
// // 	shipyard.unitHolder = holder;
// // }
