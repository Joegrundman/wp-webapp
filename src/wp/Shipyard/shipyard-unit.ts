'use strict';

class ShipyardUnit {
    public id: number;
    public x: number;
    public y: number;

    constructor (id: number, x: number, y: number) {
        this.id = id;
        this.x = x;
        this.y = y;
    }
    
}

export default ShipyardUnit;
