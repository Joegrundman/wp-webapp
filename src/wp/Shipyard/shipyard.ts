import Unit from 'Unit/unit';
import { getGame } from 'Wp/Game';
import Holder from '../UnitHolder/unitholder';
import ShipyardUnit from './shipyard-unit';

class Shipyard {

    public id: number;
    public name: string;
    public owner: string
    public rate: number;
    public unitHolder: object;
    public shipyardUnits: ShipyardUnit[];
    public dragging: boolean;
    public currentSquareX: number;
    public currentSquareY: number;
    public syContext: CanvasRenderingContext2D | null

    constructor (id: number, name: string, owner: string, rate: number) {
 
     	this.id = id;
        this.name = name;
        this.owner = owner;
        this.rate = rate;
        this.unitHolder = {};
        this.shipyardUnits = [];
        this.currentSquareX = -1;
        this.currentSquareY = -1;
        this.dragging = false;
        this.syContext = null;
    }
  
    public addShipyardUnit (shipyardUnit: ShipyardUnit) {
        this.shipyardUnits.push(shipyardUnit)
    }
   
    public removeUnitFrom (unit: ShipyardUnit) {
        this.shipyardUnits = this.shipyardUnits.filter(syunit => syunit !== unit);
    }

    public updateShipyardUnitAddress (unit: Unit) {
        // const shipyard: Shipyard = game.getShipyardFromUnit(unit.id);

        for (const shipyardUnit of this.shipyardUnits) {
            if(unit.holderX && unit.holderY && shipyardUnit.id === unit.id) {
                shipyardUnit.x = unit.holderX;
                shipyardUnit.y = unit.holderY;
            }
        }
        
    }
    /**
     * Upon selecting a shipyard, this function draws a new flag and updates details
     * @param {number} id -  the id of the shipyard
     */    
    // public handleShipyardSelected (id) {
    //     var shipyard = game.getShipyards(id);
    //     var ctyName = shipyard.owner;
    //     var cty = game.getCountryFromName(ctyName);
    //     var rate = shipyard.rate;
    //     var capacity;
    //     if (((shipyard.name == "Atlantic") || (shipyard.name == "Pacific")) && (rate <= 5)) { capacity = 10; }
    //     else { capacity = rate * 2; }
    //     $("#syCountryFlag").attr("src", WP.Country.UI.getFlagUrl(cty));
    //     $("#syRate").html(rate);
    //     $("#syCapacity").html(capacity);
    //     shipyard.draw();
    // }

    public changeRateUp () {
        // var shipyard = game.getShipyards(id);
        this.rate++;
        // shipyard.handleShipyardSelected(id);
    }
     
    public changeRateDown () {
        // var shipyard = game.getShipyards(id);
        this.rate--;
        // shipyard.handleShipyardSelected(id);        
    }
    /**
     * Upon clicking the mouse button, this function finds the unit in the shipyard display under the mouse cursor
     * @param {object} e -  the mouse event object
     */     
    // public onMouseDown (e) {
    //     var point = getPoint('shipyardCanvas', e);
    //     var stack = shipyard.unitHolder.findStackFor(point.x, point.y);
    //     if (stack) {
    //         game.setSelectedUnit(stack.getTopUnit());
    //         shipyard.unitHolder.drawStack(stack);
    //         shipyard.dragging = true;
    //     }
    //     else {
    //         game.setSelectedUnit(null);
    //     }
    // }
    /**
     * Upon double clicking the mouse, this function rotates the units in the stack under the mouse cursor
     */     
    // public onDoubleClick () {
    //     var unit = game.selectedUnit;
    //     var stack = shipyard.unitHolder.findStackContaining(unit);
    //     if (!stack) { return; }
    //     game.setSelectedUnit(null);
    //     if (stack.units.length > 1) {
    //         stack.rotateUnits();
    //         shipyard.unitHolder.drawStack(stack);
    //     }        
    // }
    /**
     * Upon moving the mouse, this function drags the currently selected unit
     * @param {object} e - the mouse event object
     */     
    // public onMouseMove (e) {
    //     shipyard.setCurrentSquare(e);
    //     if (game.selectedUnit  && shipyard.dragging == true) {
    //         var unit = game.selectedUnit;
    //         if ((shipyard.currentSquareX != unit.holderX) || (shipyard.currentSquareY != unit.holderY)) {
    //             shipyard.moveUnit(unit);
    //         }
    //     }
    // }
    /**
     * Upon letting go of the mouse button, this function stops dragging the currently selected unit
     */     
    public onMouseUp () {
        this.dragging = false
    }
    /**
     * Upon moving the mouse, this function finds the shipyard square currently under the cursor
     * @param {object} e - the mouse event object
     */   
    // public setCurrentSquare (e) {
    //     var point = getPoint('shipyardCanvas', e);
    //     this.currentSquareX = Math.floor((point.x - 10) / 66);
    //     this.currentSquareY = Math.floor((point.y - 30) / 66);
    //     if (this.currentSquareY < 0) this.currentSquareY = 0;
    //     if (this.currentSquareX < 0) this.currentSquareX = 0;
    //     if ((this.currentSquareX < 1) && (this.currentSquareY < 5)) this.currentSquareX = 1;        
    // }
    /**
     *  Updates the unit address to match its location after being dragged
     * @param {object} unit - the unit that has been moved
     */    
    // public moveUnit (unit) {
    //     var stack = shipyard.unitHolder.findStackContaining(unit);
    //     stack.removeUnit(unit);
    //     unit.holderX = shipyard.currentSquareX;
    //     unit.holderY = shipyard.currentSquareY;
    //     shipyard.updateShipyardUnitAddress(unit);
    //     shipyard.unitHolder.drawShipyard();
    // }
    /**
     * Draws the units onto the shipyard display
     */    
    public draw () {
        const units: Unit[] = []
        //   var holder = WP.UnitHolder.unitHolderBuilder(shipyardCtx, $("#syDetails"));
        if(!this.syContext) { return }

        const holder: Holder = new Holder(this.syContext);
    
        this.shipyardUnits.forEach(syUnit => {
            const unit: Unit | null = getGame().getUnitForShipyard(syUnit.id, syUnit.x, syUnit.y)
            if (unit){ units.push(unit) }
        })
        
        holder.units = units;
        holder.drawShipyard();
        this.unitHolder = holder;
    }
}

export default Shipyard;







// WP.Shipyard = function () {
// 	this.id = null;
// 	this.name = null;
// 	this.owner = null;
// 	this.rate = null;
// 	this.unitHolder = null;
// 	this.shipyardUnits = new Array();
// 	this.currentSquareX = null;
// 	this.currentSquareY = null;
// 	this.dragging = false;
// }

// WP.Shipyard.Util = {

// 	shipyardBuilder: function (id, name, owner, rate) {
// 		var shipyard = new WP.Shipyard();
// 		shipyard.id = id;
// 		shipyard.name = name;
// 		shipyard.owner = owner;
// 		shipyard.rate = rate;
// 		return shipyard;
// 	}
    
// }

// WP.Shipyard.prototype.addShipyardUnit = function (shipyardUnit) {
// 	this.shipyardUnits.push(shipyardUnit);
// 	//shipyardUnit.owner = this;
// }

// // WP.Shipyard.prototype.removeUnitFromShipyard = function (shipyard, unit) {
// // 	shipyard = game.getShipyardFromUnit(unit.id);
// // 	var j = 0;
// // 	while (j < shipyard.shipyardUnits.length) {
// // 		if (shipyard.shipyardUnits[j].id == unit.id) {
// // 			shipyard.shipyardUnits.splice(j, 1);
// // 		}
// // 		else {
// // 			j++;
// // 		}
// // 	}

// // }

// WP.Shipyard.prototype.removeUnitFrom = function (shipyard, unit) {
// 	shipyard = game.getShipyardFromUnit(unit.id);
// 	var j = 0;
// 	while (j < shipyard.shipyardUnits.length) {
// 		if (shipyard.shipyardUnits[j].id == unit.id) {
// 			shipyard.shipyardUnits.splice(j, 1);
// 		}
// 		else {
// 			j++;
// 		}
// 	}

// }

// WP.Shipyard.prototype.updateShipyardUnitAddress = function (unit) {
// 	var shipyard = game.getShipyardFromUnit(unit.id);
// 	for (var i = 0; i < shipyard.shipyardUnits.length; i++) {
// 		if (shipyard.shipyardUnits[i].id == unit.id) {
// 			shipyard.shipyardUnits[i].x = unit.holderX;
// 			shipyard.shipyardUnits[i].y = unit.holderY;
// 		}
// 	}
//     // var syUnits = game.getShipyardFromUnit(unit.id).shipyardUnits
//     // syUnits.forEach(su => {
//     //     if (su.id == unit.id) {
//     //         su.x = unit.holderX
//     //         su.y = unit.holderY
//     //     }
//     // })
// }


// WP.ShipyardUnit = function () {
// 	this.id = null;
// 	this.x = null;
// 	this.y = null;
// }

// WP.ShipyardUnit.Util = {

// 	shipyardUnitBuilder: function (id, x, y) {
// 		var shipyardUnit = new WP.ShipyardUnit();
// 		shipyardUnit.id = id;
// 		shipyardUnit.x = x;
// 		shipyardUnit.y = y;
// 		return shipyardUnit;
// 	}

// }
