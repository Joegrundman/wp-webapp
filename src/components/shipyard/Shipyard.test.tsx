import { shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import Shipyard from './Shipyard'

jest.mock('Wp/Game', () => ({
  getGame: jest.fn(() => ({
    setSelectedShipyard: jest.fn()
  }))
}))

jest.spyOn(document, 'getElementById').mockImplementation(() => {
  return {
    getBoundingClientRect: jest.fn(() => ({x: 20, y: 30})),
    getContext: jest.fn(() => ({
      beginPath: jest.fn(),
      closePath: jest.fn(),
      drawImage: jest.fn(),
      fillText: jest.fn(),
      lineTo: jest.fn(),
      moveTo: jest.fn(),
      stroke: jest.fn()
    }))
  }
})

describe('<Shipyard />', (): void => {
  describe('@render', (): void => {
    it('should render as expected with default props', () => {
      const component: ShallowWrapper = shallow(<Shipyard />)
      expect(component).toMatchSnapshot()
    })
  })

  describe('methods', (): void => {
    describe('decrementCurrentShipyard', (): void => {
      it('should decrement the currentShipyard', () => {
        const component: ShallowWrapper = shallow(<Shipyard />)
        const instance = component.instance() as Shipyard
        instance.setState({ currentShipyard: 5 })
        instance.decrementCurrentShipyard()
        expect(instance.state.currentShipyard).toBe(4)
      })

      it('should cycle the currentShipyard to the end of the list if at zero', () => {
        const component: ShallowWrapper = shallow(<Shipyard />)
        const instance = component.instance() as Shipyard
        instance.setState({ currentShipyard:  0})
        instance.decrementCurrentShipyard()
        expect(instance.state.currentShipyard).toBe(12)
      })
    })

    describe('incrementCurrentShipyard', (): void => {
      it('should increment the currentShipyard', () => {
        const component: ShallowWrapper = shallow(<Shipyard />)
        const instance = component.instance() as Shipyard
        instance.setState({ currentShipyard: 5 })
        instance.incrementCurrentShipyard()
        expect(instance.state.currentShipyard).toBe(6)
      })

      it('should cycle the currentShipyard to the start of the list if at the end', () => {
        const component: ShallowWrapper = shallow(<Shipyard />)
        const instance = component.instance() as Shipyard
        instance.setState({ currentShipyard:  12})
        instance.incrementCurrentShipyard()
        expect(instance.state.currentShipyard).toBe(0)
      })
    })
  })

})
