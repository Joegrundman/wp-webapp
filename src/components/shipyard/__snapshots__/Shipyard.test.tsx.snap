// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shipyard /> @render should render as expected with default props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Shipyard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className={undefined}
        >
          Shipyards
        </h1>,
        <div
          className={undefined}
        >
          <ArrowButton
            action={[Function]}
            direction="left"
          />
          <h2
            className={undefined}
          >
            Atlantic
          </h2>
          <ArrowButton
            action={[Function]}
            direction="right"
          />
        </div>,
        <div
          className={undefined}
          style={
            Object {
              "width": 300,
            }
          }
        >
          <canvas
            className={undefined}
            height={470}
            id="shipyardCanvasBackground"
            width={300}
          />
          <canvas
            className={undefined}
            height={470}
            id="shipyardCanvas"
            width={300}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Shipyards",
          "className": undefined,
        },
        "ref": null,
        "rendered": "Shipyards",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ArrowButton
              action={[Function]}
              direction="left"
            />,
            <h2
              className={undefined}
            >
              Atlantic
            </h2>,
            <ArrowButton
              action={[Function]}
              direction="right"
            />,
          ],
          "className": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "action": [Function],
              "direction": "left",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Atlantic",
              "className": undefined,
            },
            "ref": null,
            "rendered": "Atlantic",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "action": [Function],
              "direction": "right",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <canvas
              className={undefined}
              height={470}
              id="shipyardCanvasBackground"
              width={300}
            />,
            <canvas
              className={undefined}
              height={470}
              id="shipyardCanvas"
              width={300}
            />,
          ],
          "className": undefined,
          "style": Object {
            "width": 300,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "height": 470,
              "id": "shipyardCanvasBackground",
              "width": 300,
            },
            "ref": null,
            "rendered": null,
            "type": "canvas",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": undefined,
              "height": 470,
              "id": "shipyardCanvas",
              "width": 300,
            },
            "ref": null,
            "rendered": null,
            "type": "canvas",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className={undefined}
          >
            Shipyards
          </h1>,
          <div
            className={undefined}
          >
            <ArrowButton
              action={[Function]}
              direction="left"
            />
            <h2
              className={undefined}
            >
              Atlantic
            </h2>
            <ArrowButton
              action={[Function]}
              direction="right"
            />
          </div>,
          <div
            className={undefined}
            style={
              Object {
                "width": 300,
              }
            }
          >
            <canvas
              className={undefined}
              height={470}
              id="shipyardCanvasBackground"
              width={300}
            />
            <canvas
              className={undefined}
              height={470}
              id="shipyardCanvas"
              width={300}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Shipyards",
            "className": undefined,
          },
          "ref": null,
          "rendered": "Shipyards",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ArrowButton
                action={[Function]}
                direction="left"
              />,
              <h2
                className={undefined}
              >
                Atlantic
              </h2>,
              <ArrowButton
                action={[Function]}
                direction="right"
              />,
            ],
            "className": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "action": [Function],
                "direction": "left",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Atlantic",
                "className": undefined,
              },
              "ref": null,
              "rendered": "Atlantic",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "action": [Function],
                "direction": "right",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <canvas
                className={undefined}
                height={470}
                id="shipyardCanvasBackground"
                width={300}
              />,
              <canvas
                className={undefined}
                height={470}
                id="shipyardCanvas"
                width={300}
              />,
            ],
            "className": undefined,
            "style": Object {
              "width": 300,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "height": 470,
                "id": "shipyardCanvasBackground",
                "width": 300,
              },
              "ref": null,
              "rendered": null,
              "type": "canvas",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": undefined,
                "height": 470,
                "id": "shipyardCanvas",
                "width": 300,
              },
              "ref": null,
              "rendered": null,
              "type": "canvas",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
